tests/test_charm.py

```diff
@@ -64,6 +64,7 @@ def test_password_is_required_when_username_is_set(self):
         expected = ['prometheus-image-password']
         self.assertEqual(missing, expected)
 
+    @unittest.skip('This test will not work until operator bug #452 is fixed')
```

Comment:
Do we need to skip these tests? It looks like `altertmanager_relation_changed` doesn't need to check if `event.unit` is None because that handler is no longer looking at unit data via `event.relation.data[event.unit]`.

Reply:
@balbirthomas @chipaca 
Do either of you know what could cause event.unit to be None in production? I'm hesitant to remove the check before knowing.

Reply:
@dstathis 
This is a test to check that

1. Before an alertmanager relation is added the alerting config is empty
2. After an alertmanager relation is added there exists an alerting configuration in the prometheus pod specification

I think it is important that such a test should be fixed and active but not skipped. Would you agree ?

It is not entirely clear to me why you are hesitant to remove the check as @justinmclark seems to suggest. Why do you need the unit not to be None ? What do you think are the consequences of unit being None when a new alertmanager relation is added ? Could these consequences mitigated by appropriate behaviour in relation departed and relation broken event handlers ?